app/models/video.rb

```diff
@@ -21,11 +21,37 @@ class Video < ApplicationRecord
   before_save :set_slug
   after_save :refresh_sitemap
 
+  def provider
+    return :vimeo if vimeo?  
+    return :youtube if youtube?
+  end
+
+  def provider_id
+    send("#{provider}_id") if provider
```

Comment:
Safer to go through `attributes["#{provider}_id"]` no ?

Reply:
Ah nvm just realised the `youtube_id` is a method, I'd still advocate against using `send` and rather use a ternary or a config hash

Reply:
indeed